tests/test_content_smart_proxy.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+
+@pytest.fixture
+def content_smart_proxy(api, organization):
+    smart_proxies = api.resource('smart_proxies').call('index')
+    content_proxy = {}
+
+    for proxy in smart_proxies['results']:
+        for feature in proxy['features']:
+            if feature['name'] == 'Pulp Node':
+                content_proxy = proxy
+                break
+
+    content_proxy_id = content_proxy['id']
+
+    environments = api.resource('capsule_content').call('lifecycle_environments', {
+        'id': content_proxy['id']
+    })['results']
+
+    for environment in environments:
+        api.resource('capsule_content').call('remove_lifecycle_environment', {
+            'id': content_proxy_id,
+            'environment_id': environment['id']
+        })
+
+    library = api.resource('lifecycle_environments').call('index', {
+        'search': 'Library',
+        'organization_id': organization['id']
+    })['results'][0]
+
+    api.resource('capsule_content').call('add_lifecycle_environment', {
+        'id': content_proxy_id,
+        'environment_id': library['id']
+    })
+
+    return content_proxy
+
+
+@pytest.fixture
+def organization(api, entities):
+    return api.resource('organizations').call('index', {
+        'search': entities['organization_label']
+    })['results'][0]
+
+
+def test_smart_proxy_content_sync(api, content_smart_proxy):
```

Comment:
This test will fail on a vanilla Foreman where Pulp is not present. That's something we need to deal with in a generic way. We'll see similar things with other plugins.

If we take REX: if present, it should be tested. If it isn't, it should be skipped.

That does leave the situation where a plugin should be present. If it's skipped, we need a way to fail in case it's missing. If we take the plugins testing pipeline, we should have some way to say "expect at least these plugins to be present or fail".